mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/transformation/mpsToModelix/initial/NodeSynchronizer.kt

```diff
@@ -101,17 +110,30 @@ class NodeSynchronizer(
         mpsConcept.containmentLinks.forEach { containmentLink ->
             mpsNode.getChildren(containmentLink).forEach { mpsChild ->
                 val childLink = MPSChildLink(containmentLink)
+
+                // duplicate check
+                throwExceptionIfChildExists(cloudNode, childLink, mpsChild)
```

Comment:
Yes, you're right that in case of Nodes it's very difficult to reach that two MPS instances generate the same node ID which may conflict with each other in the cloud. So in case of Nodes, maybe it does not make sense to check for duplication.

In that case the sync is aborted and is left in an inconsistent state, which is unfortunate. User notification is still an open TODO, how we do it nicely. See [MODELIX-725](https://issues.modelix.org/issue/MODELIX-725).

Reply:
user notification is fixed in e44171b, 581db31

Reply:
I created an issue to solve the inconsistent state problem: [MODELIX-882](https://issues.modelix.org/issue/MODELIX-882)